core/dbt/adapters/base/relation.py

```diff
@@ -244,6 +244,28 @@ def create_from_node(
             **kwargs,
         )
 
+    @classmethod
+    def create_from_relation_name(cls: Type[Self], quoting: dict, relation_name: str) -> Self:
+        include_policy = cls.get_default_include_policy()
+        quote_char = cls.create().quote_character
+        parts = [part.strip(quote_char) for part in relation_name.split(".")]
+
+        database = schema = identifier = None
+
+        if include_policy["database"]:
+            database = parts.pop(0)
+        if include_policy["schema"]:
+            schema = parts.pop(0)
+        if include_policy["identifier"]:
+            identifier = parts.pop(0)
+
+        return cls.create(
+            database=database,
+            schema=schema,
+            identifier=identifier,
+            quote_policy=quoting,
+        )
```

Comment:
I could see an argument for passing in the relation_name string directly from the publication artifact if it meant less processing hops in core (e.g. relation_name from upstream manifest -> relation_name in publication artifact -> relation_name used to create a Relation). But given we still need to unpack the relation name, I don't see many benefits beyond more explicit readability in the publication artifact files (which arguably aren't supposed to be human-friendly)

A few pros in favor of using the database, schema, and alias components 
- using an existing (tested and battle-tested) method (create_from_node)
- individual components would be available for user-space code when accessing a public_node
- individual components would be more flexible for potential core implementations (validation, error messaging for example)

Downsides could be - a larger publication artifact file.. any others?

Reply:
side implementation note - we'd need to add PublicNode to `ResultNode` in nodes.py to pass a public node to `Relation.create_from`. Should we do that either way?

Reply:
I'd prefer the database/schema/identifier terminology instead of alias, because when constructing relations the field is called "identifier".

Reply:
Adding PublicNode to ResultNode would give mypy fits. I think we probably don't want to do that. We'd have to end up creating dummy properties for almost every single field in a ModelNode.